raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1039,6 +1054,14 @@ private FetchResponseData tryCompleteFetchRequest(
         }
     }
 
+    private static boolean isPartitionTruncated(FetchResponseData.PartitionData partitionResponseData) {
+        FetchResponseData.EpochEndOffset divergingEpoch = partitionResponseData.divergingEpoch();
+        FetchResponseData.SnapshotId snapshotId = partitionResponseData.snapshotId();
+
+        return divergingEpoch.epoch() != -1 || divergingEpoch.endOffset() != -1 ||
```

Comment:
nit: maybe it's just me, but it seems more intuitive to separate the diverging epoch and snapshot id checks.

Reply:
Done.